opal/mca/memory/patcher/configure.m4

```diff
@@ -45,7 +45,7 @@ AC_DEFUN([MCA_opal_memory_patcher_CONFIG],[
     memory_patcher_have___syscall_prototype=0
 
     AC_MSG_CHECKING([for __curbrk symbol])
-    AC_LINK_IFELSE([AC_LANG_PROGRAM([extern char *__curbrk;],[char *tmp = __curbrk;])],
+    AC_LINK_IFELSE([AC_LANG_PROGRAM([extern char *__curbrk;],[if (*__curbrk) return 1;])],
```

Comment:
i do not know why standard macros are used.

but why do you think printf is safer ?
the test is compiled and linked, it is never ran.
using printf means we should #include <stdio.h> to make picky compilers happy, and we have to hope it dones not pull a conflicting definition of __curbrk